teamcity: failed tests on master: acceptance/TestRapidRestarts/runMode=local, acceptance/TestRapidRestarts, acceptance/TestRapidRestarts/runMode=local, acceptance/TestRapidRestarts

The following tests appear to have failed:

[#425291](https://teamcity.cockroachdb.com/viewLog.html?buildId=425291):


```
--- FAIL: acceptance/TestRapidRestarts/runMode=local (0.000s)
Test ended in panic.

------- Stdout: -------
CockroachDB node starting at 2017-11-30 17:05:29.461724577 +0000 UTC (took 0.4s)
build:               CCL v1.2-alpha.20170901-2095-g56685a6 @ 2017/11/30 16:56:28 (go1.9)
admin:               http://127.0.0.1:46209
sql:                 postgresql://root@127.0.0.1:41143?application_name=cockroach&sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestRapidRestarts/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster764358812/1/cockroach-temp177122475
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster764358812/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster764358812/1
status:              initialized new cluster
clusterID:           d4151861-9d3d-4e33-bdbd-392f780a65df
nodeID:              1
Note: a second interrupt will skip graceful shutdown and terminate forcefully
initiating graceful shutdown of server
server drained and shutdown completed
Note: a second interrupt will skip graceful shutdown and terminate forcefully
initiating graceful shutdown of server
too early to drain; used hard shutdown instead
Note: a second interrupt will skip graceful shutdown and terminate forcefully
initiating graceful shutdown of server
too early to drain; used hard shutdown instead
Note: a second interrupt will skip graceful shutdown and terminate forcefully
initiating graceful shutdown of server
too early to drain; used hard shutdown instead
Note: a second interrupt will skip graceful shutdown and terminate forcefully
initiating graceful shutdown of server
too early to drain; used hard shutdown instead
Note: a second interrupt will skip graceful shutdown and terminate forcefully
initiating graceful shutdown of server
too early to drain; used hard shutdown instead
CockroachDB node starting at 2017-11-30 17:05:32.260853376 +0000 UTC (took 0.4s)
build:               CCL v1.2-alpha.20170901-2095-g56685a6 @ 2017/11/30 16:56:28 (go1.9)
admin:               http://127.0.0.1:34187
sql:                 postgresql://root@127.0.0.1:45903?application_name=cockroach&sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestRapidRestarts/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster854298549/1/cockroach-temp970301942
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster854298549/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster854298549/1
status:              initialized new cluster
clusterID:           8dc0e2bd-ec1b-497c-8b96-49410784e1b8
nodeID:              1
Note: a second interrupt will skip graceful shutdown and terminate forcefully
initiating graceful shutdown of server
server drained and shutdown completed
CockroachDB node starting at 2017-11-30 17:05:34.113039126 +0000 UTC (took 0.4s)
build:               CCL v1.2-alpha.20170901-2095-g56685a6 @ 2017/11/30 16:56:28 (go1.9)
admin:               http://127.0.0.1:44457
sql:                 postgresql://root@127.0.0.1:39129?application_name=cockroach&sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestRapidRestarts/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster598515600/1/cockroach-temp133830107
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster598515600/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster598515600/1
status:              initialized new cluster
clusterID:           20190634-f914-4359-8df4-68be6cd403e8
nodeID:              1
Note: a second interrupt will skip graceful shutdown and terminate forcefully
initiating graceful shutdown of server
server drained and shutdown completed
CockroachDB node starting at 2017-11-30 17:05:35.125967999 +0000 UTC (took 0.7s)
build:               CCL v1.2-alpha.20170901-2095-g56685a6 @ 2017/11/30 16:56:28 (go1.9)
admin:               http://127.0.0.1:35083
sql:                 postgresql://root@127.0.0.1:34197?application_name=cockroach&sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestRapidRestarts/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster598515600/1/cockroach-temp259133721
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster598515600/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster598515600/1
status:              restarted pre-existing node
clusterID:           20190634-f914-4359-8df4-68be6cd403e8
nodeID:              1
Note: a second interrupt will skip graceful shutdown and terminate forcefully
initiating graceful shutdown of server
server drained and shutdown completed
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xa0 pc=0x12ef7c5]

goroutine 6688 [running]:
github.com/cockroachdb/cockroach/pkg/acceptance/localcluster.(*Node).waitUntilLive(0xc420c69320)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/localcluster/cluster.go:768 +0x305
github.com/cockroachdb/cockroach/pkg/acceptance/localcluster.(*Node).StartAsync.func2(0xc420c69320, 0xc42001cfc0, 0xc42145d380)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/localcluster/cluster.go:668 +0x2b
created by github.com/cockroachdb/cockroach/pkg/acceptance/localcluster.(*Node).StartAsync
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/localcluster/cluster.go:667 +0x124
--- FAIL: acceptance/TestRapidRestarts (0.000s)
Test ended in panic.

--- FAIL: acceptance/TestRapidRestarts/runMode=local (0.000s)
Test ended in panic.

------- Stdout: -------
CockroachDB node starting at 2017-11-30 17:05:29.461724577 +0000 UTC (took 0.4s)
build:               CCL v1.2-alpha.20170901-2095-g56685a6 @ 2017/11/30 16:56:28 (go1.9)
admin:               http://127.0.0.1:46209
sql:                 postgresql://root@127.0.0.1:41143?application_name=cockroach&sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestRapidRestarts/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster764358812/1/cockroach-temp177122475
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster764358812/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster764358812/1
status:              initialized new cluster
clusterID:           d4151861-9d3d-4e33-bdbd-392f780a65df
nodeID:              1
Note: a second interrupt will skip graceful shutdown and terminate forcefully
initiating graceful shutdown of server
server drained and shutdown completed
Note: a second interrupt will skip graceful shutdown and terminate forcefully
initiating graceful shutdown of server
too early to drain; used hard shutdown instead
Note: a second interrupt will skip graceful shutdown and terminate forcefully
initiating graceful shutdown of server
too early to drain; used hard shutdown instead
Note: a second interrupt will skip graceful shutdown and terminate forcefully
initiating graceful shutdown of server
too early to drain; used hard shutdown instead
Note: a second interrupt will skip graceful shutdown and terminate forcefully
initiating graceful shutdown of server
too early to drain; used hard shutdown instead
Note: a second interrupt will skip graceful shutdown and terminate forcefully
initiating graceful shutdown of server
too early to drain; used hard shutdown instead
CockroachDB node starting at 2017-11-30 17:05:32.260853376 +0000 UTC (took 0.4s)
build:               CCL v1.2-alpha.20170901-2095-g56685a6 @ 2017/11/30 16:56:28 (go1.9)
admin:               http://127.0.0.1:34187
sql:                 postgresql://root@127.0.0.1:45903?application_name=cockroach&sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestRapidRestarts/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster854298549/1/cockroach-temp970301942
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster854298549/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster854298549/1
status:              initialized new cluster
clusterID:           8dc0e2bd-ec1b-497c-8b96-49410784e1b8
nodeID:              1
Note: a second interrupt will skip graceful shutdown and terminate forcefully
initiating graceful shutdown of server
server drained and shutdown completed
CockroachDB node starting at 2017-11-30 17:05:34.113039126 +0000 UTC (took 0.4s)
build:               CCL v1.2-alpha.20170901-2095-g56685a6 @ 2017/11/30 16:56:28 (go1.9)
admin:               http://127.0.0.1:44457
sql:                 postgresql://root@127.0.0.1:39129?application_name=cockroach&sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestRapidRestarts/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster598515600/1/cockroach-temp133830107
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster598515600/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster598515600/1
status:              initialized new cluster
clusterID:           20190634-f914-4359-8df4-68be6cd403e8
nodeID:              1
Note: a second interrupt will skip graceful shutdown and terminate forcefully
initiating graceful shutdown of server
server drained and shutdown completed
CockroachDB node starting at 2017-11-30 17:05:35.125967999 +0000 UTC (took 0.7s)
build:               CCL v1.2-alpha.20170901-2095-g56685a6 @ 2017/11/30 16:56:28 (go1.9)
admin:               http://127.0.0.1:35083
sql:                 postgresql://root@127.0.0.1:34197?application_name=cockroach&sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestRapidRestarts/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster598515600/1/cockroach-temp259133721
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster598515600/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster598515600/1
status:              restarted pre-existing node
clusterID:           20190634-f914-4359-8df4-68be6cd403e8
nodeID:              1
Note: a second interrupt will skip graceful shutdown and terminate forcefully
initiating graceful shutdown of server
server drained and shutdown completed
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xa0 pc=0x12ef7c5]

goroutine 6688 [running]:
github.com/cockroachdb/cockroach/pkg/acceptance/localcluster.(*Node).waitUntilLive(0xc420c69320)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/localcluster/cluster.go:768 +0x305
github.com/cockroachdb/cockroach/pkg/acceptance/localcluster.(*Node).StartAsync.func2(0xc420c69320, 0xc42001cfc0, 0xc42145d380)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/localcluster/cluster.go:668 +0x2b
created by github.com/cockroachdb/cockroach/pkg/acceptance/localcluster.(*Node).StartAsync
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/localcluster/cluster.go:667 +0x124
--- FAIL: acceptance/TestRapidRestarts (0.000s)
Test ended in panic.


```

Please assign, take a look and update the issue accordingly.

--- a/pkg/acceptance/localcluster/cluster.go
+++ b/pkg/acceptance/localcluster/cluster.go
@@ -737,11 +737,14 @@ func (n *Node) waitUntilLive() {
 		Multiplier:     2,
 	}
 	for r := retry.Start(opts); r.Next(); {
+		var pid int
 		n.Lock()
-		exited := n.cmd == nil
+		if n.cmd != nil {
+			pid = n.cmd.Process.Pid
+		}
 		n.Unlock()
-		if exited {
-			return
+		if pid == 0 {
+			return // process already quit
 		}
 
 		urlBytes, err := ioutil.ReadFile(n.listeningURLFile())
@@ -765,7 +768,6 @@ func (n *Node) waitUntilLive() {
 		pgURL.Path = n.Cfg.DB
 		n.Lock()
 		n.pgURL = pgURL.String()
-		pid := n.cmd.Process.Pid
 		n.Unlock()
 
 		var uiURL *url.URL

